src/proj7BittingCerratoCohenEllmer/bantam/parser/Parser.java

```diff
@@ -638,97 +643,109 @@ private Expr parseUnaryPrefix() {
     // <VarExpr> ::= <VarExprPrefix> <Identifier> <VarExprSuffix>
     // <VarExprPrefix> ::= SUPER . | THIS . | EMPTY
     // <VarExprSuffix> ::= ( <Arguments> ) | EMPTY
     private Expr parsePrimary() {
-        int startPosition = currentToken.position;
-        Expr expr;
-        ExprList args;
+        int position = currentToken.position;
+        Expr exprToReturn;
+
         // handle ( <Expression> )
-        if(currentToken.kind == Token.Kind.LPAREN){
-            currentToken = scanner.scan(true); // @<Expression>
-            expr = parseExpression(); // @ )
-            ensureTokenType("Incomplete Expression: Unclosed Parenthesis", Token.Kind.RPAREN);
+        if (currentToken.kind == Token.Kind.LPAREN) {
+            currentToken = scanner.scan(true); // currentToken: <Expression>
+            exprToReturn = parseExpression(); // currentToken: )
+            ensureTokenType("Incomplete Expression: Unclosed Parenthesis",
+                    Token.Kind.RPAREN);
             currentToken = scanner.scan(true); // prep next token
-        //handle integerConst
-        }else if(currentToken.kind == Token.Kind.INTCONST){
-            expr = parseIntConst(); // @ next token -- throws error
-        //handle booleanConst
-        }else if(currentToken.kind == Token.Kind.BOOLEAN){
-            expr = parseBoolean(); // @ next token -- throws error
-        // handle StringConst
-        }else if(currentToken.kind == Token.Kind.STRCONST){
-            expr = parseStringConst(); // @ next token -- throws error
-        // handle VarExpr
-        } else{// @<VarExprPrefix>::= SUPER . | THIS . | EMPTY
-            Expr refExpr = null;
-            String methodName;
-            if("super".equals(currentToken.spelling)
-                || "this".equals(currentToken.spelling)){
-
-                refExpr = parseExpression(); //currentToken -> '.'
-                ensureTokenType("reference call missing seperator '.'", Token.Kind.DOT);
-
-                currentToken = scanner.scan(true); // '.' -> <identifier>
+
+            // handle integerConst
+        } else if (currentToken.kind == Token.Kind.INTCONST) {
+            exprToReturn = parseIntConst();
+
+            // handle booleanConst
+        } else if (currentToken.kind == Token.Kind.BOOLEAN) {
+            exprToReturn = parseBoolean();
+
+            // handle StringConst
+        } else if (currentToken.kind == Token.Kind.STRCONST) {
+            exprToReturn = parseStringConst();
+
+            // handle VarExpr
+        } else {
+            // get prefix (super or this)
+            Expr refExpr;
+            String spelling = currentToken.spelling;
+            if ("super".equals(spelling) || "this".equals(spelling)) {
+                // todo: TEST THE SHIT OUT OF THIS
+                refExpr = parseExpression(); // currentToken: '.'
+                ensureTokenType("reference call missing separator '.'", Token.Kind.DOT);
```

Comment:
i am concerned if super.super() is a valid call will super() also be a valid call... this might throw an error expecting a '.'